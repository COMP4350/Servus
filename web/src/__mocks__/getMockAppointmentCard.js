function getMockAppointmentCard() {
    return {
        _id: '123',
        buyer: 'Risto',
        provider: 'Arveend',
        service_id: '1001',
        created_at: new Date(100),
        booked_time: new Date(500),
    };
}

export const mockAppointmentCard = getMockAppointmentCard();
export const mockAppointments = [getMockAppointmentCard()];
