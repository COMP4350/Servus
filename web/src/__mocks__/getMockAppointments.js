function getMockAppointments() {
    return [
        {
            buyer: 'Risto',
            provider: 'Arveend',
            service_id: '1001',
            created_at: new Date(100),
            booked_time: new Date(500),
        },
        {
            buyer: 'Arveend',
            provider: 'Risto',
            service_id: '1002',
            created_at: new Date(700),
            booked_time: new Date(1200),
        },
        {
            buyer: 'Caden',
            provider: 'Andy',
            service_id: '1003',
            created_at: new Date(1400),
            booked_time: new Date(1900),
        },
    ]
}

function getMockAppointmentsCookie() {
    return {
        username: 'testusername',
    }
}

export const mockAppointments = getMockAppointments();
export const mockAppointmentsCookie = getMockAppointmentsCookie();