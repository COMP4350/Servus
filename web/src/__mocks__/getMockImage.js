function getMockImage() {
    return {
        ownerUsername: 'riki',
        uploadDate: new Date(2012, 1, 1),
        imageName: 'example_image.jpeg',
        imageData: 'the/link/to/data',
        profilePicture: false,
    };
}

export const mockImage = getMockImage();
