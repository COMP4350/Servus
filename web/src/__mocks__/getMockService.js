function getMockService() {
    return {
        _id: '1001',
        name: "Risto's Painting",
        provider: 'Risto',
        description: 'Risto paints you',
        cost: 100,
        duration: '0100',
        availability: [{ weekday: 1, start_time: '0800', end_time: '1800' }],
        ratings: [{ username: 'arvind', rating: 5 }],
    };
}

export const mockService = getMockService();
