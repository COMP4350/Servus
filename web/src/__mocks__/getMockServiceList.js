function getMockServiceList() {
    return [
        {
            name: "Risto's Painting",
            provider: 'Risto',
            description: 'Risto paints you',
            cost: 100,
            duration: 60,
        },
        {
            name: "Reesto's Painting",
            provider: 'Reesto',
            description: 'Reesto paints yee',
            cost: 120,
            duration: 80,
        },
        {
            name: "Rasta's Painting",
            provider: 'Rasta',
            description: 'Rasta paints yaa',
            cost: 150,
            duration: 90,
        },
    ];
}

export const mockServiceList = getMockServiceList();
