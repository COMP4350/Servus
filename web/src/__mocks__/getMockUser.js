function getMockUser() {
    return {
        username: 'testuser',
        firstName: 'test',
        lastName: 'user',
        password: '123',
        bio: 'im a test user',
    };
}

export const mockUser = getMockUser();
