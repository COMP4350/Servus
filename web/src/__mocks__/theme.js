import { createMuiTheme } from '@material-ui/core/';

export default createMuiTheme({
    palette: {
        primary: {
            main: '#647AA3',
        },
        secondary: {
            main: '#EC5732',
        },
        default: {
            main: '#FFF2EB',
        },
    },
    background: {
        dark: '#151515',
        main: '#272727',
    },
    typographs: {
        fontFamily: ['Roboto'],
    },
});
