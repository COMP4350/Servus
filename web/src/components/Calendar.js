import React, { useState, useEffect } from 'react';
import axios from 'axios';
import TableCell from '@material-ui/core/TableCell';
import { darken, fade, lighten } from '@material-ui/core/styles/colorManipulator';
import Typography from '@material-ui/core/Typography';
import { EditingState, ViewState } from '@devexpress/dx-react-scheduler';
import classNames from 'clsx';
import {
    Scheduler,
    MonthView,
    Appointments,
    Toolbar,
    DateNavigator,
    AppointmentTooltip,
    AppointmentForm,
    EditRecurrenceMenu,
    Resources,
    DragDropProvider,
} from '@devexpress/dx-react-scheduler-material-ui';
import { withStyles } from '@material-ui/core/styles';

const getBorder = theme => (`1px solid ${theme.palette.type === 'light'
    ? lighten(fade(theme.palette.divider, 1), 0.88)
    : darken(fade(theme.palette.divider, 1), 0.68)
    }`);

const DayScaleCell = props => (
    <MonthView.DayScaleCell {...props} style={{ textAlign: 'center', fontWeight: 'bold' }} />
);

const styles = theme => ({
    cell: {
        color: '#78909C',
        position: 'relative',
        userSelect: 'none',
        verticalAlign: 'top',
        padding: 0,
        height: 100,
        borderLeft: getBorder(theme),
        '&:first-child': {
            borderLeft: 'none',
        },
        '&:last-child': {
            paddingRight: 0,
        },
        'tr:last-child &': {
            borderBottom: 'none',
        },
        '&:hover': {
            backgroundColor: 'white',
        },
        '&:focus': {
            backgroundColor: fade(theme.palette.primary.main, 0.15),
            outline: 0,
        },
    },
    content: {
        display: 'flex',
        justifyContent: 'center',
        width: '100%',
        height: '100%',
        position: 'absolute',
        alignItems: 'center',
    },
    text: {
        padding: '0.5em',
        textAlign: 'center',
    },
    opacity: {
        opacity: '0.5',
    },
    appointment: {
        borderRadius: '10px',
        '&:hover': {
            opacity: 0.6,
        },
    },
    apptContent: {
        '&>div>div': {
            whiteSpace: 'normal !important',
            lineHeight: 1.2,
        },
    },
    flexibleSpace: {
        color: '#78909C',
        flex: 'none',
    },
    flexContainer: {
        display: 'flex',
        alignItems: 'center',
    },
    tooltipContent: {
        padding: theme.spacing(3, 1),
        paddingTop: 0,
        backgroundColor: theme.palette.background.paper,
        boxSizing: 'border-box',
        width: '400px',
    },
    tooltipText: {
        ...theme.typography.body2,
        display: 'inline-block',
    },
    title: {
        ...theme.typography.h6,
        color: theme.palette.text.secondary,
        fontWeight: theme.typography.fontWeightBold,
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        whiteSpace: 'nowrap',
    },
    icon: {
        color: theme.palette.action.active,
        verticalAlign: 'middle',
    },
    circle: {
        width: theme.spacing(4.5),
        height: theme.spacing(4.5),
        verticalAlign: 'super',
    },
    textCenter: {
        textAlign: 'center',
    },
    dateAndTitle: {
        color: '#78909C',
        lineHeight: 1.1,
    },
    titleContainer: {
        paddingBottom: theme.spacing(2),
    },
    container: {
        paddingBottom: theme.spacing(1.5),
    },
});

// #FOLD_BLOCK
const CellBase = React.memo(({
    classes,
    startDate,
    formatDate,
    otherMonth,
    // #FOLD_BLOCK
}) => {
    const isFirstMonthDay = startDate.getDate() === 1;
    const formatOptions = isFirstMonthDay
        ? { day: 'numeric', month: 'long' }
        : { day: 'numeric' };
    return (
        <TableCell
            tabIndex={0}
            className={classNames({
                [classes.cell]: true,
                [classes.opacity]: otherMonth,
            })}
        >
            <div className={classes.content}>
            </div>
            <div className={classes.text}>
                {formatDate(startDate, formatOptions)}
            </div>
        </TableCell>
    );
});

const TimeTableCell = withStyles(styles, { name: 'Cell' })(CellBase);

const Appointment = withStyles(styles, { name: 'Appointment' })(({ classes, ...restProps }) => (
    <Appointments.Appointment
        {...restProps}
        className={classes.appointment}
    />
));

const AppointmentContent = withStyles(styles, { name: 'AppointmentContent' })(({ classes, ...restProps }) => (
    <Appointments.AppointmentContent {...restProps} className={classes.apptContent} />
));

const FlexibleSpace = withStyles(styles, { name: 'ToolbarRoot' })(({ classes, ...restProps }) => (
    <Toolbar.FlexibleSpace {...restProps} className={classes.flexibleSpace}>
        <div className={classes.flexContainer}>
            <Typography variant="h5" style={{ marginLeft: '10px' }}>My Calendar</Typography>
        </div>
    </Toolbar.FlexibleSpace>
));

const getRandomColor = () => {
    var letters = '0123456789ABCDEF';
    var color = '#';
    for (var i = 0; i < 6; i++) {
        color += letters[Math.floor(Math.random() * 16)];
    }
    return color;
}
let setColor = false;

const Calendar = ({ appointments }) => {

    const [data, setData] = useState([]);
    const [owners, setOwners] = useState([]);
    const newDate = new Date();

    const commitDeletes = async ({ deleted }) => {
        const response = await axios.delete(`/appointment/${deleted}`);
        console.log(`im the response: ${response}`);
        setData(data.filter(apt => apt._id !== deleted));
    }

    const passInfo = () => {
        setOwners([]);
        setData([]);
        appointments?.forEach((apt, i) => {
            const newOwner = {
                text: apt.provider,
                id: i,
                color: getRandomColor(),
            }
            setOwners(owners => [...owners, newOwner]);
            axios.get(`/services/${apt.service_id}`)
                .then(response => {
                    const service = response.data.result;
                    const startDate = new Date(apt.booked_time);
                    const endDate = new Date(startDate);
                    endDate.setTime(endDate.getTime() + service.duration * 60 * 1000);
                    const tempObj = {
                        id: apt._id,
                        title: service.name,
                        startDate: startDate,
                        endDate: endDate,
                        ownerId: i,
                    }
                    setData(data => [...data, tempObj]);
                }
                );
        });
        setColor = false;
    }
    if (!setColor && appointments.length === owners.length) {
        owners?.forEach(name => {
            for (let i = 1; i < owners.length; i++) {
                if (name.text === owners[i].text && owners[i].color !== name.color) {
                    owners[i].color = name.color;
                }
            }
        });
        setColor = true;
    }
    const resources = [{
        fieldName: 'ownerId',
        title: 'Owners',
        instances: owners,
    }];

    useEffect(() => {
        passInfo();
    }, [])

    return (
        <Scheduler
            data={data}
        >
            <EditingState
                onCommitChanges={commitDeletes}
            />
            <ViewState
                defaultCurrentDate={newDate}
            />

            <MonthView
                timeTableCellComponent={TimeTableCell}
                dayScaleCellComponent={DayScaleCell}
            />

            <Appointments
                appointmentComponent={Appointment}
                appointmentContentComponent={AppointmentContent}
            />
            <Resources
                data={resources}
            />

            <Toolbar
                flexibleSpaceComponent={FlexibleSpace}
            />
            <DateNavigator />

            <EditRecurrenceMenu />
            <AppointmentTooltip
                showCloseButton
                showDeleteButton
                showOpenButton
            />
            <AppointmentForm />
            <DragDropProvider />
        </Scheduler>
    );
}

export default Calendar;