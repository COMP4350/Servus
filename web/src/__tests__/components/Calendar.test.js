import React from 'react';
import { act, render } from '@testing-library/react';
import Calendar from '../../components/Calendar';
import { mockAppointments } from '../../__mocks__/getMockAppointmentCard';
import { mockService } from '../../__mocks__/getMockService';

import mockAxios from 'axios';

describe('Calendar', () => {
    it('renders correctly', async () => {
        const promise = Promise.resolve();
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: { result: mockService },
            })
        );

        const { asFragment } = render(
            <Calendar appointments={mockAppointments} />
        );
        expect(mockAxios.get).toHaveBeenCalledWith('/services/1001');
        expect(mockAxios.get).toHaveBeenCalledTimes(1);
        expect(asFragment()).toMatchSnapshot();

        await act(() => promise);
    });
});
