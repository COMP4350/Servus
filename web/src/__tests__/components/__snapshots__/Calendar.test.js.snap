// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Root-container-2"
    style="height: 100%;"
  >
    <div
      class="OverlayContainer-container-3"
    />
    <div
      class="OverlayContainer-container-3"
    />
    <div
      class="MuiToolbar-root MuiToolbar-regular Toolbar-toolbar-13 MuiToolbar-gutters"
    >
      <div
        class="RootBase-root-14"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root NavigationButton-button-15"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root NavigationButton-button-15"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text OpenButton-textButton-16"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            April 2021
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root OpenButton-iconButton-17"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="FlexibleSpace-flexibleSpace-35 ToolbarRoot-flexibleSpace-24"
      >
        <div
          class="ToolbarRoot-flexContainer-25"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5"
            style="margin-left: 10px;"
          >
            My Calendar
          </h5>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-container-36"
    >
      <div>
        <div
          class="makeStyles-stickyElement-37 makeStyles-header-38 makeStyles-flexRow-45"
        >
          <div
            class="makeStyles-background-48 makeStyles-inlineFlex-47 makeStyles-ordinaryHeaderBorder-42"
          >
            <div>
              <table
                class="MuiTable-root makeStyles-table-50 makeStyles-table-51"
              >
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Sun
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Mon
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Tue
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Wed
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Thu
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Fri
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-52"
                      style="text-align: center; font-weight: bold;"
                    >
                      <div
                        class="Cell-dayOfWeek-53"
                      >
                        Sat
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-flexRow-45"
        >
          <div
            class="makeStyles-inlineFlex-47"
          >
            <div
              class="makeStyles-relativeContainer-46"
            >
              <table
                class="MuiTable-root makeStyles-table-50 makeStyles-table-55"
              >
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        28
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        29
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        30
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        31
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        April 1
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        2
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        3
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        4
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        5
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        6
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        7
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        8
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        9
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        10
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        11
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        12
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        13
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        14
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        15
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        16
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        17
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        18
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        19
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        20
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        21
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        22
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        23
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        24
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        25
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        26
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        27
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        28
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        29
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        30
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        May 1
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        2
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        3
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        4
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        5
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        6
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        7
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body Cell-cell-56 Cell-opacity-59"
                      tabindex="0"
                    >
                      <div
                        class="Cell-content-57"
                      />
                      <div
                        class="Cell-text-58"
                      >
                        8
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="AppointmentsContainer-container-73"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
